### ISSUE TITLE
Address review: BINF-D-17-00305-R1

### ISSUE THREAD

**Issue by juanmirocks** [Rostlab/LocText#39]
BINF-D-17-00305
LocText: relation extraction of protein localizations to assist database curation
Juan Miguel Cejuela; Shrikant Vinchurkar; Tatyana Goldberg; Madhukar Sollepura Prabhu Shankar; Ashish Baghudana; Aleksander Bojchevski; Carsten Uhlig; André Ofner; Pandu Raharja-Liu; Lars Juhl Jensen; Burkhard Rost
BMC Bioinformatics

Dear M. Sc. Cejuela,

Your manuscript "LocText: relation extraction of protein localizations to assist database curation" (BINF-D-17-00305) has been assessed by our reviewers. They have raised a number of points which we believe would improve the manuscript and may allow a revised version to be published in BMC Bioinformatics.

Their reports, together with any other comments, are below. Please also take a moment to check our website at http://binf.edmgr.com/ for any additional comments that were saved as attachments.

If you are able to fully address these points, we would encourage you to submit a revised manuscript to BMC Bioinformatics.
Once you have made the necessary corrections, please submit online at:

http://binf.edmgr.com/

If you have forgotten your username or password please use the "Send Login Details" link to get your login information. For security reasons, your password will be reset.

Please include a cover letter with a point-by-point response to the comments, describing any additional experiments that were carried out and including a detailed rebuttal of any criticisms or requested revisions that you disagreed with. Please also ensure that all changes to the manuscript are indicated in the text by highlighting or using track changes.

Please also ensure that your revised manuscript conforms to the journal style, which can be found in the Instructions for Authors on the journal homepage.

A decision will be made once we have received your revised manuscript, which we expect by 30 Jul 2017.

Please note that you will not be able to add, remove, or change the order of authors once the editor has accepted your manuscript for publication.  Any proposed changes to the authorship must be requested during peer-review, and adhere to our criteria for authorship as outlined in BioMed Central's policies.  To request a change in authorship, please download the 'Request for change in authorship form' which can be found here - http://www.biomedcentral.com/about/editorialpolicies#authorship.  Please note that incomplete forms will be rejected.  Your request will be taken into consideration by the editor, and you will be advised whether any changes will be permitted.  Please be aware that we may investigate, or ask your institute to investigate, any unauthorized attempts to change authorship or discrepancies in authorship between the submitted and revised versions of your manuscript.

I look forward to receiving your revised manuscript and please do not hesitate to contact us if you have any questions.

Best wishes,

Rezarta Islamaj Dogan, Ph.D.
BMC Bioinformatics
https://bmcbioinformatics.biomedcentral.com/

Editor Comments:

The reviewers have raised some valid concerns, both in terms of language and methodology description, and other important details that could benefit your manuscript. I look forward to the revision,

Reviewer reports:

Reviewer 1: The paper presents a method for extracting protein location information by text-mining publications. The achieved performance is objectively quite low (partially because of variability and ambiguity of protein naming in literature), but enough to be a valuable support tool for an expert.

I have some concerns, though:

* [x] Restructure sections somehow -- 

> The text is hard to follow, especially given BMC standards; main insights are tightly intermixed with implementation details. I would advise to re-structure text into a methods-at-the-end order.

* [x] Support the decision of SVMs --> how does it fear against naive Bayes or logistic regression?
* [x] Provide more details on feature selection
  * ⚠️ Note that I barely changed the text. I did renamed the section from "Features" to "Feature selection"
* [x] Put all features in SOM

>Machine learning methodology is valid, but I would appreciate some better discussion of other method performance -- if non-linear SVM does not improve accuracy, it suggests some trivial method like naive Bayes or logistic regression may be equally good (while not requiring hyper-parameters). In a sharp contrast to the rest of the paper, there are no details about feature selection, which is likely the most crucial part of modelling, given that data is wide and there is a lot of correlated features. The supplement should also present the whole set of final features rather than just examples.

* ~~[ ] Generate ROC curve~~ > _PR should be more relevant since we care about precision_
* ~~[ ] Generate PR curve (PRC)~~. ([Btw](https://stats.stackexchange.com/questions/7207/roc-vs-precision-and-recall-curves))
* ~~[ ] Showcase what best reflects the difference between LocText and Baseline~~

--> ⚠️ Left untouched and explained in text.

> - I am not convinced by the comparison of LocText with the "Baseline" approach; F-measure is statistically the same for both methods, which may suggest that differences in precision and recall are just due to a different focus and not really meaningful. A ROC or PR-curve analysis would clear this up.

---

Reviewer 2: This study presents the construction and application of LocText, a text mining based tool for the extraction of protein localization information from the biomedical literature. LocText consists of a fully automated pipeline for name-entity recognition (NER), relation extraction (RE) and linking to the original data sources.

The paper provides a clear explanation of the crucial steps to address for identifying instances, link them to database identifiers and find relationships between them. It also presents one careful explanation of the methods employed in this work and a comparative analysis with other approaches although some aspect of the evaluation process are not entirely clear.

Comments

* [x] Clarify this sentence:
> 1- Background section "The native sub-cellular localization or location of a protein is an important aspect of its function."
Can the authors expand a bit on this concept. It is a bit too concise and leaves too much to intuition why protein localization is crucial for protein function. What native means exactly in this context?

* [x] Update reference
> 2- Reference 2 is very old, please update

* [x] Review reference
> 3- Reference 3 is a predictor not a database based on experimental GO annotation

--> Perhaps:
  * http://www.nature.com.eaccess.ub.tum.de/nmeth/journal/v12/n9/full/nmeth.3555.html#ref1
  * http://www.mcponline.org/content/13/8/1937
  * https://elifesciences.org/articles/16950

* [x] Somehow address
* [x] Answer why GO is the ontology fit.
> 4- Is GO the best reference dictionary for protein localization? Are there any other resources available? Is the granularity of GO cellular component an advantage or disadvantage for the proposed task?

* [x] Address somehow data size
  --> We just say that we only had access to it.
* [x] Clarify that, yes, LocText was used for the intrinsic evaluation of the method
* [x] Clarify the difference of [intrinsic vs extrinsic evaluation](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2517250/)

> 5- The LocTextCorpus consists of 100 abstracts. Is it sized appropriately for the scope? In other terms, has it been used as it is because it was the only available resource or because it met all the required criteria? Was it used only for training purpose or also in the evaluation?

* [x] Clarify validation: intrinsic LocText + PubMed manual evaluation
> 6- I have not clear how the validation was conducted. Were the PubMed queries the test set? Using simple PubMed queries doesn't introduce a strong bias in the evaluation process? How the predictions have been validated? Manually? Did you check in the negative set to identify false negatives? Can you generate a table with the absolute numbers that details the results by test set (different organisms)?

* [x] Address
> 7- Please remove qualifications from Fig3 and Table1.
Fig3 LocText full pipeline (NER + RE), high precision -> Fig3 LocText full pipeline (NER + RE) precision
LocText (RE only) and STRING Tagger (NER), highly performant ->  LocText (RE only) and STRING Tagger (NER) performance.

* [x] Address or rebut
> 8- Please limit the use of first person plural and phrasing like "encouraged".
Eg: "We evaluated relationship extraction" -> Relationship extraction was evaluated

---

Mine:

* [x] Fix Aleksandar's name
* [x] Put LocText code on tagtog page
* [x] Review back all bibliography

---

Lars:

Beginning to look at it, starting with figures:

* Figure 1 is fine as is.

* [x] In figure 2, I would drop the "D" on the x-axis labels. When the axis
is called "sentence distance", I would argue the distances are simply
0, 1, 2, 3, etc. Also, I think you need a label on the y-axis. It is
not clear from just looking at the figure what the percentages mean. I
assume the label would be something like "Percentage of
protein–localization pairs". Finally, I would drop the heading ... I
find it misleading (unless I misunderstand the figure).

* [x] The heading of Figure 3 is similarly not very informative. I would
just delete it.

Best,
Lars

---

* [x] Perhaps do PR
> (xxbr: thought we changed this view during our walk?)
--> ⚠️ just included an inline PR-curve analysis in the response letter

* [x] Review **intrinsic vs. extrinsic** naming 
> ; xxbr: again: might have to change the name
--> ⚠️ Left the same 

* [x] Check again percentages (like recall...)
